platform/src/lib/utils/url.ts

```diff
@@ -0,0 +1,75 @@
+import { parse as parseUrl, format as formatUrl, UrlObject } from 'url';
```

Comment:
__note:__ copied and TS-fied from https://github.com/elastic/kibana/blob/master/src/utils/modify_url.js